### PR TITLE
client: fix the nil interface conversion bug

### PR DESCRIPTION
**Pull request by JmPotato**
Signed-off-by: JmPotato <ghzpotato@gmail.com>

### What problem does this PR solve?

A bug found from pingcap/tidb#20980. There is a certain chance that when a TSO request is received, the corresponding DC's `tsDeadline` is not ready.

https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/tidb_ghpr_check_2/detail/tidb_ghpr_check_2/61998/pipeline/49/

### What is changed and how it works?

Manually check the `tsDeadline` when it's not ready.

### Check List

Tests

- Unit test
- Integration test

### CODE REVIEW

**Review by lhy1024 — APPROVED** [review]

File: client/client.go [hunk 1]
```diff
@@ -179,6 +179,8 @@ type client struct {
 	tsDeadline sync.Map // Same as map[string]chan deadline
 	// dc-location -> *lastTSO
 	lastTSMap sync.Map // Same as map[string]*lastTSO
+
+	checkTSDeadlineCh chan struct{}
```
Comment (HunDunDM) [inline client/client.go:183]: It seems more reasonable to use `context`.

**Review by Yisaer — APPROVED** [review]
LGTM

File: client/client.go (hunk 1 above)
Comment (Yisaer) [reply client/client.go:183]: why?

Comment (JmPotato) [reply client/client.go:183]: May I ask why? The channel is more easy to use in this case.

Comment (HunDunDM) [reply client/client.go:183]: Sorry, I read something wrong.

**Review by HunDunDM — APPROVED** [review]

**Pull request merged**